File: tests/fixtures/swift/generics.swift
class Container<T> {
    var value: T

    init(value: T) {
        self.value = value
    }

    func map<U>(transform: (T) -> U) -> Container<U> {
        return Container<U>(value: transform(value))
    }
}

protocol Comparable2 where Self: Equatable {
    func compare(other: Self) -> Int
}

struct Pair<A, B> {
    let first: A
    let second: B
}

File: tests/fixtures/scala/generics.scala
package com.example.generics

class Container[+A](val value: A) {
  def map[B](f: A => B): Container[B] = new Container(f(value))
}

trait Comparable[A <: Ordered[A]] {
  def compare(other: A): Int
}

class Pair[A, B](val first: A, val second: B)
